circuits/utils/transformTwo.ts

```diff
@@ -1,32 +1,51 @@
 import { writeFileSync } from 'fs';
-import path from 'path';
 import { Colors, Puzzles } from '../types/circuitFunctions.types';
+import path from 'path';
 const puzzles: Puzzles = require('../test/data/puzzles.json');
 
-export function transformTwo(
-  column: Array<Colors>,
-  inColor: Colors,
-  outColor: Colors
-): Array<Colors> {
-  if (column.length === 0) {
-    return column;
-  }
+type TransformTwoArguments = {
+  column: Array<Colors>;
+  inColor: Colors;
+  outColorBot: Colors;
+  outColorTop: Colors;
+};
+
+export function transformTwo({
+  column,
+  inColor,
+  outColorBot,
+  outColorTop,
+}: TransformTwoArguments): Array<Colors> {
+  if (column.length === 1 || column.length === 0) return column;
```

Comment:
so if there is 1 element left couldnt it still change colour?

Reply:
no. it is a base case, so it returns an array with one element since transformation is not permitted on the last element in order to keep return `column` length equal to `column` length on initial call.

Reply:
so thats a design decision of the transformTwo function - can only transform [inColor] to [outColorBot, outColorTop] **not** [inColor] to [outColorBot] if at last index.
I'm cool with that btw I just thought intuitively should be the latter